pkg/proxy/kafka.go

```diff
@@ -106,115 +106,113 @@ func createKafkaRedirect(r *Redirect, conf kafkaConfiguration) (RedirectImplemen
-		flowdebug.Log(log.WithField(logfields.Request, req.String()),
-			"No Kafka rules loaded, rejecting")
+		flowdebug.Log(scopedLog, "No Kafka rules matching identity, rejecting")
 		return false
 	}
 
 	b, err := json.Marshal(rules.Kafka)
 	if err != nil {
-		flowdebug.Log(log.WithError(err).WithField(logfields.Request, req.String()),
-			"Error marshalling kafka rules to apply")
+		flowdebug.Log(scopedLog, "Error marshalling kafka rules to apply")
 		return false
 	} else {
-		flowdebug.Log(log.WithFields(logrus.Fields{
-			logfields.Request: req.String(),
-			"rule":            string(b),
-		}), "Applying rule")
+		flowdebug.Log(scopedLog.WithField("rule", string(b)), "Applying rule")
 	}
 
 	return req.MatchesRule(rules.Kafka)
 }
 
 // kafkaLogRecord wraps an accesslog.LogRecord so that we can define methods with a receiver
 type kafkaLogRecord struct {
-	accesslog.LogRecord
-	redirect *Redirect
-	req      *kafka.RequestMessage
+	logger.LogRecord
+	topics []string
 }
 
-func (k *kafkaRedirect) newKafkaLogRecord(req *kafka.RequestMessage) kafkaLogRecord {
-	record := kafkaLogRecord{
-		LogRecord: req.GetLogRecord(),
-		req:       req,
-		redirect:  k.redirect,
+func apiKeyToString(apiKey int16) string {
+	if key, ok := api.KafkaReverseAPIKeyMap[apiKey]; ok {
+		return key
 	}
+	return fmt.Sprintf("%d", apiKey)
+}
 
-	if k.redirect.ingress {
-		record.ObservationPoint = accesslog.Ingress
-	} else {
-		record.ObservationPoint = accesslog.Egress
+func (k *kafkaRedirect) newLogRecordFromRequest(req *kafka.RequestMessage) kafkaLogRecord {
+	return kafkaLogRecord{
+		LogRecord: logger.NewLogRecord(k.redirect, accesslog.TypeRequest,
+			logger.LogTags.Kafka(&accesslog.LogRecordKafka{
+				APIVersion:    req.GetVersion(),
+				APIKey:        apiKeyToString(req.GetAPIKey()),
+				CorrelationID: int32(req.GetCorrelationID()),
+			})),
+		topics: req.GetTopics(),
 	}
-
-	return record
 }
 
-func (l *kafkaLogRecord) fillInfo(r *Redirect, srcIPPort, dstIPPort string, srcIdentity uint32) {
-	fillInfo(r, &l.LogRecord, srcIPPort, dstIPPort, srcIdentity)
-}
+func (k *kafkaRedirect) newLogRecordFromResponse(res *kafka.ResponseMessage, req *kafka.RequestMessage) kafkaLogRecord {
+	lr := kafkaLogRecord{
+		LogRecord: logger.NewLogRecord(k.redirect, accesslog.TypeResponse,
+			logger.LogTags.Kafka(&accesslog.LogRecordKafka{})),
+	}
 
-// log Kafka log records
-func (l *kafkaLogRecord) log(typ accesslog.FlowType, verdict accesslog.FlowVerdict, code int, info string) {
-	l.Type = typ
-	l.Verdict = verdict
-	l.Kafka.ErrorCode = code
-	l.Info = info
-	l.Timestamp = time.Now().UTC().Format(time.RFC3339Nano)
+	if res != nil {
+		lr.Kafka.CorrelationID = int32(res.GetCorrelationID())
+	}
 
-	l.LogRecord.NodeAddressInfo = accesslog.NodeAddressInfo{
-		IPv4: node.GetExternalIPv4().String(),
-		IPv6: node.GetIPv6().String(),
+	if req != nil {
+		lr.Kafka.APIVersion = req.GetVersion()
+		lr.Kafka.APIKey = apiKeyToString(req.GetAPIKey())
+		lr.topics = req.GetTopics()
```

Comment:
Ditto, should we be filling clientID in response if present in corresponding request